pkg/controller/argocd/secret.go

```diff
@@ -423,12 +440,22 @@ func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.Arg
 	}
 	for _, s := range clusterSecrets.Items {
 		// check if cluster secret with default server address exists
-		// do nothing if exists.
+		// update the list of namespaces if value differs.
 		if string(s.Data["server"]) == common.ArgoCDDefaultServer {
 			if clusterConfigInstance {
-				r.client.Delete(context.TODO(), &s)
+				if err := r.client.Delete(context.TODO(), &s); err != nil {
+					return err
+				}
 			} else {
-				return nil
+				ns := strings.Split(string(s.Data["namespaces"]), ",")
+				for _, n := range namespaces {
+					if !containsString(ns, strings.TrimSpace(n)) {
+						ns = append(ns, strings.TrimSpace(n))
+					}
+					sort.Strings(ns)
+					s.Data["namespaces"] = []byte(strings.Join(ns, ","))
+					return r.client.Update(context.TODO(), &s)
```

Comment:
I don't think we want to return yet.  Also, we should only need to call `Update` once outside of the for-loop.

Reply:
Technically, there would always be only one instance of the Secret for the default cluster server, but if there are multiple instances (which is highly unlikely) we would update the first available instance. 
Do you think we should be trying to update all the secrets if there are multiple instances?

Reply:
We do assume (and support) only one ArgoCD instance per namespace, so I think it's ok.